### PR TITLE
#4 wr 52.taqc python test project 

### PR DESCRIPTION
**Pull request by Thoughtseize1**
Here's a brief overview of the changes I made:

Added Locators: I have included new locators to improve the element identification in the code. 

**Added Test Cases:**

1. test_registration_form_appears_after_click: This test case verifies that the registration form appears correctly after a specific action (a button click).
2. b) test_sign_in_with_right_credentials: This test case checks the functionality of the sign-in process using correct credentials. It validates that users can successfully log in with their accurate information.
3. c) test_sign_up_with_incorrect_data: This test case ensures that the sign-up process handles incorrect data appropriately. It simulates a scenario where users provide incorrect information during the sign-up process and verifies that the system handles it correctly.

I have carefully migrated these test cases from my old code and have written them following our established coding style and best practices.

Please review the changes and provide any feedback or suggestions you may have. I appreciate your time and attention to this pull request.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -128,4 +129,12 @@ dmypy.json
...
-.DS_Store
\ No newline at end of file
+.DS_Store
+.idea/inspectionProfiles/profiles_settings.xml
+.idea/inspectionProfiles/Project_Default.xml
+.idea/misc.xml
+.idea/modules.xml
+.idea/toolchains.xml
+.idea/vcs.xml
+.idea/Wr-52.TAQC-Python-TestProject.iml
```
Comment (servettonga) [inline .gitignore:139]: Adding ".idea/" should cover them all

File: src/pages/ModalPage.py [hunk 2]
```diff
@@ -19,6 +19,7 @@ class ModalPage(BaseWrapper):
     FORM_REGISTER_PASSWORD_REPEAT_INP_XPATH = f"{MODAL_DIALOG_XPATH}//input[@name='RepeatPassword']"
     FORM_BTN_XPATH = f"{MODAL_DIALOG_XPATH}//span[@class='MuiButton-label']"
     SUCCESS_PAGE_ALERT_TEXT_CSS = "div.alert-success"
+    UNSUCCESS_PAGE_ALERT_TEXT_XPATH = "/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[3]/div/p"
```
Comment (servettonga) [inline src/pages/ModalPage.py:22]: Try with this instead: `//input[@name='password']/parent::*/following-sibling::p[contains(@class, 'Mui-error')]`

File: src/resources/locators.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    REGISTRATION_FORM = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]")
+    FIELD_SIGN_IN_ENTER_EMAIL = (By.XPATH, '/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[1]/div/div/input')
+    FIELD_SIGN_IN_ENTER_PASSWORD = (By.XPATH, '/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[2]/div/div/input')
+    FIELD_SIGN_UP_FOR_EMAIL = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[1]/div/div/input")
+    FIELD_SIGN_UP_FOR_PASSWORD = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[2]/div/div/input")
+    FIELD_SIGN_UP_FOR_REPEAT_PASSWORD = (
+    By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[3]/div/div/input")
+    WRONG_DATA = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[3]/div/p")
+    BUTTON_SIGN_UP_CONFIRM = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[4]/div/button[2]/span[1]")
+
```
Comment (servettonga) [inline src/resources/locators.py:42]: Try to optimize selectors with more reliable and readable ones.

File: tests/testScripts/test_landing.py [hunk 4]
```diff
@@ -1,7 +1,10 @@
 # It'll be the importing of report tools here
 
 import config
+import time
 import pytest
+from src.resources.config_credentials import irina_email, irina_password, irina_account_name, wrong_email, \
```
Comment (servettonga) [inline tests/testScripts/test_landing.py:6]: You can simplify the whole credential process with a different approach.

**Review by servettonga — COMMENTED** [review]
Tests pass.
Comments added for files individually.

File: src/resources/config_credentials.py
Comment (mehalyna) [inline src/resources/config_credentials.py:1]: Looks good. But better to wrap up this code into the function. And use return for the output

File: src/resources/locators.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        "#root > div > div > article:nth-child(1) > div > div > button",
+    )
+    BUTTON_REGISTER = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[1]/div/div/button[2]/span[1]")
+    BUTTON_FIND_EVENT = (By.XPATH, r'//*[@id="root"]/div/div/article[1]/div/div/a')
+    BUTTON_SIGN_IN_UP = (By.ID, "headbtn")
+    BUTTON_WITH_LOGO = (By.ID, "EEButton")
+    BUTTON_SIGN_IN = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[3]/div/button[2]/span[1]")
+    BUTTON_JOIN_EVENTS_EXPRESS_ = (
+        By.XPATH,
+        r'//*[@id="root"]/div/div/article[2]/div[3]/div/button',
```
Comment (mehalyna) [inline src/resources/locators.py:18]: eg. BUTTON_JOIN_EVENTS_EXPRESS
//*[@id="root"]//*[contains(text(),'Join EventsExpress')]/..
looks better for

File: src/resources/locators.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+class HeadersLocators(object):
+    ACCOUNT_NAME = (By.XPATH, '//*[@id="userNameAlign"]')
+
+
+class RegistrationFormLocators(object):
+    REGISTRATION_FORM = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]")
+    FIELD_SIGN_IN_ENTER_EMAIL = (By.XPATH, '/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[1]/div/div/input')
+    FIELD_SIGN_IN_ENTER_PASSWORD = (By.XPATH, '/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[2]/div/div/input')
+    FIELD_SIGN_UP_FOR_EMAIL = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[1]/div/div/input")
+    FIELD_SIGN_UP_FOR_PASSWORD = (By.XPATH, "/html/body/div[3]/div[3]/div/div/div[2]/div/form/div[2]/div/div/input")
```
Comment (vtelt) [inline src/resources/locators.py:37]: The long XPATHes can be broken after some changes of UI and it is hard to update it all time. So it's good idea to use more short, for example, with "contains": //button/span[contains(text(), "Sign Up")]

File: .gitignore (hunk 1 above)
Comment (Thoughtseize1) [reply .gitignore:139]: Leave only .idea/

File: src/pages/ModalPage.py (hunk 2 above)
Comment (Thoughtseize1) [reply src/pages/ModalPage.py:22]: Yeah. It works - thanks

File: src/resources/locators.py (hunk 3 above)
Comment (Thoughtseize1) [reply src/resources/locators.py:42]: I removed that file cause we decide to describe locators in pages

File: tests/testScripts/test_landing.py (hunk 4 above)
Comment (Thoughtseize1) [reply tests/testScripts/test_landing.py:6]: Done. Added credentials to config.py

File: src/resources/config_credentials.py
Comment (Thoughtseize1) [reply src/resources/config_credentials.py:1]: I just moved it to config.py. Should i leave imy credentials  there or move yourd admin credentials onto my code and write it into function?

File: src/resources/locators.py (hunk 5 above)
Comment (Thoughtseize1) [reply src/resources/locators.py:18]: I removed that file cause we decide to describe locators in pages

File: src/resources/locators.py (hunk 6 above)
Comment (Thoughtseize1) [reply src/resources/locators.py:37]: I removed that file cause we decide to describe locators in pages

**Review by mehalyna — CHANGES_REQUESTED** [review]
what about my comments?

**Review by mehalyna — APPROVED** [review]

**Review by vtelt — APPROVED** [review]

**Pull request merged**